android/src/main/kotlin/com/signify/hue/flutterreactiveble/channelhandlers/ScanDevicesHandler.kt

```diff
@@ -33,19 +33,21 @@ class ScanDevicesHandler(private val bleClient: com.signify.hue.flutterreactiveb
     }
 
     private fun startDeviceScan() {
-            scanParameters?.let { params ->
-                scanForDevicesDisposable = bleClient.scanForDevices(params.filter, params.mode)
-                        .observeOn(AndroidSchedulers.mainThread())
-                        .subscribe(
-                        { scanResult ->
-                            handleDeviceScanResult(converter.convertScanInfo(scanResult))
-                        },
-                        { throwable ->
-                            Timber.d("Error while scanning for devices:  ${throwable.message}")
-                            handleDeviceScanResult(converter.convertScanErrorInfo(throwable.message))
-                        }
-                )
-            } ?: handleDeviceScanResult(converter.convertScanErrorInfo("Scanparameters are not being set"))
+        scanParameters?.let { params ->
+            scanForDevicesDisposable = bleClient.scanForDevices(params.filter, params.mode, params.locationServiceIsMandatory)
+                    .observeOn(AndroidSchedulers.mainThread())
+                    .subscribe(
+                            { scanResult ->
+                                handleDeviceScanResult(converter.convertScanInfo(scanResult))
+                            },
+                            { throwable ->
+
+                                Timber.d("Error while scanning for devices:  ${throwable.message}")
```

Comment:
Why pollute the log? The caller will know about the failure, because the exception is propagated to them (right?).

As I mentioned before, my opinion is that a library should no pollute the log. At all. The library users will not (normally) debug it, therefore such logging will just complicate their life. For us it's not needed either, unless we are specifically debugging library-level a BLE issue, in which case it's not a problem to add the relevant logging.

I understand it was there before. Perhaps, we should clean the logging in the Android implementation up. At least that would be mine proposition.